docs/naps/0-nap-process.md

```diff
@@ -0,0 +1,298 @@
+---
+Author: "Juan Nunez-Iglesias <mailto:jni@fastmail.com>"
+Author: "Andy Sweet <mailto:andrewdsweet@gmail.com>"
+Author: "Kevin Yamauchi <mailto:kevin.yamauchi@gmail.com>"
+Created: '2022-03-23'
+Status: Active
+Type: Process
+---
+
+(nap0)=
+
+# NAP 0 — Purpose and Process
+
+## What is a NAP?
+
+NAP stands for Napari Advancement Proposal. A NAP is a design document
+providing information to the community, or describing a new feature for
+napari or its processes or environment. The NAP should provide a rationale
+for the proposed change as well as a concise technical specification, if
+applicable.
+
+We intend NAPs to be the primary mechanisms for proposing major new
+features, for collecting community input on an issue, and for documenting
+the design decisions that have gone into napari. The NAP author is
+responsible for building consensus within the community and documenting
+dissenting opinions.
+
+Because the NAPs are maintained as text files in a versioned repository,
+their revision history is the historical record of the feature proposal
+[^id3].
+
+### When is a NAP warranted?
+
+Most contributions to napari should go through the standard [contributing
+process](napari-contributing), that is, opening a pull request to the main
+repository. They will typically be uncontroversial improvements, and
+require little design discussion. The git commit and the pull request
+itself serve as an adequate record of the contribution's history.
+
+In some cases, contributions will require extensive discussions around any
```

Comment:
I'd more broadly recommend a quarterly/halfly review of roadmap intentions, user experience expectations, and working session of outstanding assumptions/unknowns about the users and their experience. This sort of working session (which I'm happy/excited to run!) could allow us to hear from the napari core developer community and do some research ahead of potential NAP ideas.

Reply:
@LCObus I love these ideas, but honestly in my opinion NAPs are the *output* of such sessions. ie if you want to run a user experience survey thingy, run it, make an issue with the results, and then if anything in that issue requires it, make a NAP. Does that make sense?

Reply:
It does! My questions are about the happenings pre-NAP, that connect the input to the output.  How might we integrate user experience research into a community sourced NAP system? In other words:
1. How might we build a collective roadmap that socializes bigger community questions so that those questions can be supported by rigorous user experience research? (A roadmap would also mitigate potential redundant NAPs)
3. How might someone with a potential NAP but outstanding exploratory questions socialize those questions in a way that they could be prioritized into a study that supports their eventual NAP?

Reply:
They are very good questions but imho questions that should be resolved after this PR, not during this PR. We can maybe even create a NAP for pre-NAPs. 😂 Bonus points if you can come up with a backronym for "LULLABY" 😜

Reply:
love it.